tests/contrib/test_cache.py

```diff
@@ -131,6 +131,23 @@ def test_purge(self):
         # Cache purges old items *before* it sets new ones.
         assert len(c._cache) == 3
 
+    def test_no_timeout(self, c):
```

Comment:
Could you prefix those with `test_generic` instead of `test`? The above instance of `test_purge` is a mistake.